### PR TITLE
Golang runtime support

### PR DESCRIPTION
**Pull request by Johnbush**
**Description of the Change**

- Added install scripts for golang and it's dependencies

### CODE REVIEW

File: installscripts/dockerfiles/jenkins-ce/Dockerfile [hunk 1]
```diff
@@ -2,12 +2,18 @@
...
 curl https://bootstrap.pypa.io/get-pip.py | python && pip install --upgrade awscli && apt-get clean && pip install virtualenv && /usr/bin/easy_install virtualenv
 RUN wget -O /opt/apache-maven-3.5.2-bin.tar.gz https://archive.apache.org/dist/maven/maven-3/3.5.2/binaries/apache-maven-3.5.2-bin.tar.gz && tar xzvf /opt/apache-maven-3.5.2-bin.tar.gz -C  /opt && export PATH=$PATH:/opt/apache-maven-3.5.2/bin >> /etc/profile.d/maven.sh && ln -sf /opt/apache-maven-3.5.2/bin/mvn /usr/bin/mvn
 RUN curl -sL https://deb.nodesource.com/setup_8.x | bash && apt-get install -y nodejs && npm install  -global serverless@1.30.0 @angular/cli@1.7.3 jshint
+# Install scripts for golang 
+RUN curl -O https://storage.googleapis.com/golang/go${goVersion}.linux-amd64.tar.gz && tar -xvf go${goVersion}.linux-amd64.tar.gz && mv go /usr/local && \
+echo 'export GOROOT=/usr/local/go' | sudo tee -a /etc/profile  && echo 'export PATH=$PATH:/usr/local/go/bin' | sudo tee -a /etc/profile && source /etc/profile
+# Install godep dependency Management tool
+RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (rajeevr2715) [inline installscripts/dockerfiles/jenkins-ce/Dockerfile]: Please put the commands in a single RUN command if possible.
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/

**Comment by rajeevr2715** [comment]
@bleggett Should this has to be applied in scenario1 also? ie, run the include_recipe 'golang::default' (https://supermarket.chef.io/cookbooks/golang)

**Comment by bleggett** [comment]
Yeah this will need to be added to the Chef cookbook as well (for existing Jenkins instances).

There are a few prebuilt cookbooks (https://supermarket.chef.io/cookbooks?utf8=%E2%9C%93&q=golang&platforms%5B%5D=) but they all look either unmaintained or basic, so manually fetching the runtime might be more reliable.

File: installscripts/dockerfiles/jenkins-ce/Dockerfile (hunk 1 above)
Comment (Johnbush) [reply installscripts/dockerfiles/jenkins-ce/Dockerfile]: Updated.

**Comment by Johnbush** [comment]
@bleggett I have created a custom recipe which will install golang along with it's dependency godep.

**Comment by bleggett** [comment]
@Johnbush Oh if you need to install `dep` too that's fine, that makes sense, as long as the Go stuff is a separate recipe (which it is), I just would rather avoid using handrolled recipes in general.

(I thought `vgo` was replacing `dep`?)

**Comment by devsatishm** [comment]
> Can you do that? It should be a simple change and will save us the effort next release.

I like it. @Johnbush Can we explore this?

File: installscripts/dockerfiles/jenkins-ce/Dockerfile [hunk 2]
```diff
@@ -2,12 +2,18 @@
...
 curl https://bootstrap.pypa.io/get-pip.py | python && pip install --upgrade awscli && apt-get clean && pip install virtualenv && /usr/bin/easy_install virtualenv
 RUN wget -O /opt/apache-maven-3.5.2-bin.tar.gz https://archive.apache.org/dist/maven/maven-3/3.5.2/binaries/apache-maven-3.5.2-bin.tar.gz && tar xzvf /opt/apache-maven-3.5.2-bin.tar.gz -C  /opt && export PATH=$PATH:/opt/apache-maven-3.5.2/bin >> /etc/profile.d/maven.sh && ln -sf /opt/apache-maven-3.5.2/bin/mvn /usr/bin/mvn
 RUN curl -sL https://deb.nodesource.com/setup_8.x | bash && apt-get install -y nodejs && npm install  -global serverless@1.30.0 @angular/cli@1.7.3 jshint
+# Install scripts for golang 
+RUN curl -O https://storage.googleapis.com/golang/go${goVersion}.linux-amd64.tar.gz && tar -xvf go${goVersion}.linux-amd64.tar.gz && mv go /usr/local && \
+echo 'export GOROOT=/usr/local/go' | sudo tee -a /etc/profile  && echo 'export GOBIN=$GOROOT/bin' | sudo tee -a /etc/profile &&  echo 'export GOPATH=$GOROOT/src' | sudo tee -a /etc/profile && \
+echo 'export PATH=$PATH:$GOPATH:$GOBIN' | sudo tee -a /etc/profile && source /etc/profile && RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (rajeevr2715) [inline installscripts/dockerfiles/jenkins-ce/Dockerfile]: @Johnbush Please remove the extra RUN added here.

Comment (Johnbush) [reply installscripts/dockerfiles/jenkins-ce/Dockerfile]: ok

**Pull request merged**